test/Vault.spec.ts

```diff
@@ -826,36 +1016,32 @@ describe("Vault", () => {
       await addYieldToVault("100");
       const tx = await vault.connect(carol).claimYield(carol.address);
 
+      const yieldWithFee = parseUnits("50");
+      const perfFee = yieldWithFee.mul(PERFORMANCE_FEE_PCT).div(DENOMINATOR);
+      const carolYield = await vault.yieldFor(carol.address);
+      expect(carolYield[0]).to.eq(parseUnits("0"));
+      expect(carolYield[1]).to.eq(parseUnits("0"));
+      expect(carolYield[2]).to.eq(parseUnits("0"));
+      expect(await vault.accumulatedPerfFee()).to.eq(perfFee);
+      expect(await underlying.balanceOf(carol.address)).to.eq(
+        yieldWithFee.sub(perfFee)
+      );
+      const bobYield = await vault.yieldFor(bob.address);
+      expect(bobYield[0]).to.eq(yieldWithFee.sub(perfFee));
+      expect(bobYield[1]).to.eq(parseUnits("25").mul(SHARES_MULTIPLIER));
+      expect(bobYield[2]).to.eq(perfFee);
+
       await expect(tx)
         .to.emit(vault, "YieldClaimed")
         .withArgs(
           1,
           carol.address,
-          parseUnits("50"),
-          parseUnits("25").mul(SHARES_MULTIPLIER)
+          yieldWithFee.sub(perfFee),
+          parseUnits("25").mul(SHARES_MULTIPLIER),
+          perfFee
         );
     });
 
-    it("claims the yield of a user", async () => {
```

Comment:
Why did you remove the test to emit an event? We had two tests instead of one to make it easier to debug if something was wrong.

Reply:
i think we don't need add one test case for testing event.
I added event test in next case.

Reply:
We could write all the tests in a single test case, but doing so makes it harder to read and understand what's failing. That's why I wrote two test cases, with one just for the event. Maybe only one of them fails, and that immediately sends a message about what could be wrong. The test was already there, and I don't see why you would remove it.